omod/src/main/java/org/openmrs/module/queue/web/QueueEntryMetricRestController.java

```diff
@@ -71,8 +72,12 @@ public Object handleRequest(HttpServletRequest request) {
 			ret.add(COUNT, services.getQueueEntryService().getCountOfQueueEntries(criteria).intValue());
 		} else {
 			List<QueueEntry> queueEntries = services.getQueueEntryService().getQueueEntries(criteria);
+			// Filter queueEntries where ended_at is null. The  count should be for those waiting for a service
+			List<QueueEntry> filteredQueueEntries = queueEntries.stream().filter(entry -> entry.getEndedAt() == null)
+			        .collect(Collectors.toList());
+			
```

Comment:
@makombe thanks for this PR.  This endpoint does not make any assumptions about the metric being requested, and its intentionally kept generic.  As you can see, it will run a search based on the request parameters passed into it, and then return the relevant count of those search results.

I assume this is attempting to fix invalid metric data on the frontend, in which case the simple fix would be to pass an additional parameter in the frontend request of `isEnded=false`.  You can see the possible search criteria parameters in the `QueueEntrySearchCriteriaParser.java` class.

I recommend that we close this PR and instead make this fix in the service-queues app within patient managment.